apps/designer/app/shared/sync/sync-stores.ts

```diff
@@ -0,0 +1,104 @@
+import { applyPatches } from "immer";
+import store, { type Change } from "immerhin";
+import type { ValueContainer } from "react-nano-state";
+import { useEffect } from "react";
+import { allUserPropsContainer } from "@webstudio-is/react-sdk";
+import { publish, subscribe } from "~/shared/pubsub";
+import {
+  rootInstanceContainer,
+  breakpointsContainer,
+  designTokensContainer,
+  stylesContainer,
+} from "~/shared/nano-states";
+
+type StoreData = {
+  namespace: string;
+  value: unknown;
+};
+
+declare module "~/shared/pubsub" {
+  export interface PubsubMap {
+    sendStoreData: StoreData[];
+    sendStoreChanges: {
+      // distinct target to avoid infinite loop
+      // @todo implement two-way patches
+      target: "canvas" | "designer";
+      changes: Change[];
+    };
+  }
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const containers = new Map<string, ValueContainer<any>>([
+  ["breakpoints", breakpointsContainer],
+  ["root", rootInstanceContainer],
+  ["styles", stylesContainer],
+  ["props", allUserPropsContainer],
+  ["designTokens", designTokensContainer],
+]);
```

Comment:
how can we avoid having a central place that knows about all of these?

Reply:
it was this way as well, but I think if this list supposed to grow over time, it should be decentralized, so that each container is responsible for it's registration

immerhin already throws an error if we pass a container to transaction that hasn't been registered with immerhin

Reply:
I think we couod have `registerContainer(namespace, container)`

Reply:
Maybe later. There is still no immerhin in designer.